### PR TITLE
[api-logs] Expose logging artifacts as public

### PR DESCRIPTION
**Pull request by CodeBlanch**
Relates to #4433

## Changes

* Exposes the logging artifacts in `OpenTelemetry.Api` and removes the added `InternalsVisibleTo`s

## Merge requirement checklist

* [X] [CONTRIBUTING](https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/CONTRIBUTING.md) guidelines followed (nullable enabled, static analysis, etc.)
* [X] Appropriate `CHANGELOG.md` files updated for non-trivial changes
* [ ] Changes in public API reviewed (if applicable)

### CODE REVIEW

**Review by vishweshbankwar — APPROVED** [review]

File: src/OpenTelemetry.Api/CHANGELOG.md [hunk 1]
```diff
@@ -11,6 +11,11 @@
   other than the three types mentioned.
   ([#4542](https://github.com/open-telemetry/opentelemetry-dotnet/pull/4542))
 
+* Added [Logs Bridge
+  API](https://github.com/open-telemetry/opentelemetry-specification/blob/976432b74c565e8a84af3570e9b82cb95e1d844c/specification/logs/bridge-api.md)
+  artifacts.
```
Comment (reyang) [inline src/OpenTelemetry.Api/CHANGELOG.md:16]: What does "artifact" mean here? (which part of https://en.wikipedia.org/wiki/Artifact_(software_development))

Comment (CodeBlanch) [reply src/OpenTelemetry.Api/CHANGELOG.md:16]: "Things" 🤣 I changed the entry to call out the main pieces LMK if you want it to say more.

Comment (reyang) [reply src/OpenTelemetry.Api/CHANGELOG.md:16]: I guess many folks would interpret "adding ... artifacts" as "introducing a new NuGet package or DLL". I suggest that we simply say "added logs bridge API" ("to the existing API artifact" - which can be easily deduced from the changelog context).

Comment (CodeBlanch) [reply src/OpenTelemetry.Api/CHANGELOG.md:16]: Something weird is going on with GitHub. I pushed to the branch behind this PR but the PR isn't updating. Here's the commit if you want to check out the text I'm _trying_ to use 😄 

https://github.com/CodeBlanch/opentelemetry-dotnet/commit/20e0221133ff2f164adc71cd612bb66e48ace56b

Comment (reyang) [reply src/OpenTelemetry.Api/CHANGELOG.md]: The new version is much easier to understand 👍

File: src/OpenTelemetry.Api/.publicApi/net462/PublicAPI.Unshipped.txt [hunk 2]
```diff
@@ -0,0 +1,79 @@
+abstract OpenTelemetry.Logs.Logger.EmitLog(in OpenTelemetry.Logs.LogRecordData data, in OpenTelemetry.Logs.LogRecordAttributeList attributes = default(OpenTelemetry.Logs.LogRecordAttributeList)) -> void
```
Comment (utpilla) [inline src/OpenTelemetry.Api/.publicApi/net462/PublicAPI.Unshipped.txt:1]: Did we not try to avoid APIs with default parameters?

**Pull request merged**